Specs/Renderer/FramebufferManagerSpec.js

```diff
@@ -0,0 +1,446 @@
+      expect(fbm.getDepthRenderbuffer()).toBeDefined();
+    });
+
+    it("creates renderbuffer depth attachments if supportsDepthTexture is true but context.depthTexture is false", function () {
+      fbm = new FramebufferManager({
+        depth: true,
+        supportsDepthTexture: true,
+      });
+      context._depthTexture = false;
+      fbm.update(context, 1, 1);
+      expect(fbm.getDepthTexture()).toBeUndefined();
+      expect(fbm.getDepthRenderbuffer()).toBeDefined();
+    });
+
+    it("destroys attachments and framebuffer", function () {
+      if (!context.drawBuffers) {
+        return;
+      }
+
+      fbm = new FramebufferManager({
+        colorAttachmentsLength: 2,
+        depth: true,
+      });
+      fbm.update(context, 1, 1);
+      expect(fbm.framebuffer).toBeDefined();
+      expect(fbm.getColorTexture(0)).toBeDefined();
+      expect(fbm.getColorTexture(1)).toBeDefined();
+      expect(fbm.getDepthRenderbuffer()).toBeDefined();
+
+      spyOn(Framebuffer.prototype, "destroy").and.callThrough();
+      spyOn(Renderbuffer.prototype, "destroy").and.callThrough();
+      spyOn(Texture.prototype, "destroy").and.callThrough();
+      fbm.destroy();
+
+      expect(Framebuffer.prototype.destroy.calls.count()).toEqual(1);
+      expect(fbm.framebuffer).toBeUndefined();
+      expect(Renderbuffer.prototype.destroy.calls.count()).toEqual(1);
+      expect(fbm.getDepthRenderbuffer()).toBeUndefined();
+      expect(Texture.prototype.destroy.calls.count()).toEqual(2);
+      expect(fbm.getColorTexture(0)).toBeUndefined();
+      expect(fbm.getColorTexture(1)).toBeUndefined();
+    });
+
+    it("does not destroy attachments that are not created by FramebufferManager", function () {
+      fbm = new FramebufferManager({
+        createColorAttachments: false,
+        createDepthAttachments: false,
+      });
+
+      var colorTexture = new Texture({
+        context: context,
+        width: 1,
+        height: 1,
+        pixelFormat: PixelFormat.RGBA,
+      });
+      fbm.setColorTexture(colorTexture, 0);
+
+      var depthRenderbuffer = new Renderbuffer({
+        context: context,
+        width: 1,
+        height: 1,
+      });
+      fbm.setDepthRenderbuffer(depthRenderbuffer);
+
+      fbm.update(context, 1, 1);
+      fbm.destroy();
+      expect(fbm.framebuffer).toBeUndefined();
+      expect(fbm.getColorTexture()).toBeDefined();
+      expect(fbm.getDepthRenderbuffer()).toBeDefined();
+
+      colorTexture.destroy();
+      depthRenderbuffer.destroy();
+    });
+
+    it("does not destroy resources if texture dimensions haven't changed", function () {
+      fbm = new FramebufferManager();
+      spyOn(FramebufferManager.prototype, "destroy").and.callThrough();
+      fbm.update(context, 1, 1);
+      fbm.update(context, 1, 1);
+      expect(FramebufferManager.prototype.destroy.calls.count()).toEqual(1);
+    });
+
+    it("destroys resources after texture dimensions change", function () {
```

Comment:
Worth adding a few more related tests:

* `destroys resources after pixel datatype changes`
* `destroys resources after pixel format changes`
* `destroys resources after new attachment is set`